File: 卡码/代码随想录/JS/链表/05-删除链表倒数第n个节点/LeetCode_0019.js
//19. 删除链表的倒数第 N 个结点
function ListNode(val,next){
  this.val=(val===undefined ? 0 : val);
  this.next=(next===undefined ? null : next);
}

const removeNthFromEnd = (head, n)=> {
  let dummy=new ListNode(-1,head);
  let fast=dummy,slow=dummy;
  n++;
  while(n-->0) fast=fast.next;
  while(fast!=null){
    fast=fast.next;
    slow=slow.next;
  }
  slow.next=slow.next.next;
  return dummy.next;
};

File: 卡码/代码随想录/C++/链表/05-删除链表倒数第n个节点/LeetCode_0019.cpp
//19. 删除链表的倒数第 N 个结点
struct ListNode{
	int val;
	ListNode* next;
	ListNode():val(0),next(nullptr){}
	ListNode(int val):val(val),next(nullptr){}
	ListNode(int val,ListNode* next):val(val),next(next){}
};

class Solution {
public:
  ListNode* removeNthFromEnd(ListNode* head, int n) {
		ListNode* dummy=new ListNode(-1,head);
		ListNode* fast=dummy;
		ListNode* slow=dummy;
		n++;
		while(n--) fast=fast->next;
		while(fast){
			fast=fast->next;
			slow=slow->next;
		}
		ListNode* tmp=slow->next;
		slow->next=tmp->next;
		delete tmp;
		return dummy->next;
  }
};
